src/extensions/scratch3_mqtt/index.js

```diff
@@ -303,60 +302,48 @@ class Scratch3MQTTBlocks {
 
     disconnectFromBroker(args) {
         console.log("[mqtt] disconnecting from MQTT broker... ");
-        if ( this._client != null ) {  this._client.end(); }
+        if ( this._client ) {  this._client.end(); }
     }
 
     subscribe( args ) {
+        if ( !this.isConnected() ) {
```

Comment:
Using early returns prevents nested `if`-statements and makes it a bit more readable when we're doing some validation before continuing.